multi_source_job_queue.go

```diff
@@ -37,26 +41,40 @@ func (msjq *MultiSourceJobQueue) Jobs(ctx gocontext.Context) (outChan <-chan Job
 			}).Error("failed to get job chan from queue")
 			return nil, err
 		}
-		buildJobChans[fmt.Sprintf("%s.%d", queue.Name(), i)] = jc
+		qName := fmt.Sprintf("%s.%d", queue.Name(), i)
+		buildJobChans[qName] = jc
 	}
 
 	go func() {
 		for {
 			for queueName, bjc := range buildJobChans {
+				var job Job = nil
 				jobSendBegin := time.Now()
+				logger = logger.WithField("queue_name", queueName)
+
+				logger.Debug("about to receive job")
 				select {
-				case job := <-bjc:
-					buildJobChan <- job
-					metrics.TimeSince("travis.worker.job_queue.multi.blocking_time", jobSendBegin)
-					logger.WithFields(logrus.Fields{
-						"source": queueName,
-						"dur":    time.Since(jobSendBegin),
-					}).Info("sent job to multi source output channel")
+				case job = <-bjc:
```

Comment:
I think we could keep job logging and sending inside this clause. It shouldn't change the behaviour and it feels easier to follow (at least for me).